pkg/proc/bininfo.go

```diff
@@ -2196,6 +2202,76 @@ func (bi *BinaryInfo) ListPackagesBuildInfo(includeFiles bool) []*PackageBuildIn
 	return r
 }
 
+const (
+	fakeAddressBase     = 0xbeef000000000000
+	fakeAddressUnresolv = 0xbeed000000000000 // this address never resloves to memory
+)
+
+// newCompositeMemory creates a new compositeMemory object and registers it.
+// If the same composite memory has been created before it will return a
+// cached object.
+// This caching is primarily done so that registerized variables don't get a
+// different address every time they are evaluated, which would be confusing
+// and leak memory.
+func (bi *BinaryInfo) newCompositeMemory(mem MemoryReadWriter, regs op.DwarfRegisters, pieces []op.Piece, descr *locationExpr) (int64, *compositeMemory, error) {
```

Comment:
I think this should live on `Target` or stay within `pkg/proc/mem.go`. `BinaryInfo` is intended to contain information we've parsed from the binary on disk, memory is more of a running process concern.

Reply:
Done. The outcome is less than ideal since that code path ends up getting called by the breakpoint setting code, which does not have access to the Target object.